### PR TITLE
Leave the input/output/error handles unset when they are not redirected

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

This PR is to fix `Start-Process` to address https://github.com/PowerShell/PSReadLine/issues/288.

Basically, when starting `pwsh` with `Start-Process` in the following ways, the new console and the existing console get weirdly tied together, which causes `Console.ReadKey` running in those 2 consoles to interfere with each other (block and unblock each other):
- `Start-Process pwsh -RedirectStandardError .\error.txt`: start `pwsh` with its stderr redirected to the file `error.txt`
- `Start-Process pwsh -PassThru -Credential (Get-Credential <another-user>)`: start `pwsh` as a different user.

Based on the investigation in https://github.com/PowerShell/PSReadLine/issues/288#issuecomment-1673605171, we should not explicitly set the standard input/output/error handles when they are not redirected. Instead, just let Windows figure out the default to use when creating the process.

Note that, the change in `TaskbarJumpList.cs` is related. When running `Start-Process pwsh -Credential` as a non-admin user, the "TaskbarJumpList" code throws `AccessDenied` exception, which causes a debug build to crash. The `Debug.Fail` is not needed. The `catch (ThreadStartException)` below simply ignore the exception too.

## Manual validation

I verified the changes on both Windows 11 and Windows 10, and it works fine for the abovementioned scenarios. I also verified that the `"redirecting input/output/error to files"` are still working fine, regardless of using `-Credential` or not.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs [hunk 1]
```diff
@@ -117,7 +116,6 @@ private static void CreateElevatedEntry(string title)
                     var CLSID_EnumerableObjectCollection = new Guid(@"2d3468c1-36a7-43b6-ac24-d3f02fd9607a");
                     const uint CLSCTX_INPROC_HANDLER = 2;
                     const uint CLSCTX_INPROC = CLSCTX_INPROC_SERVER | CLSCTX_INPROC_HANDLER;
-                    var ComSvrInterface_GUID = new Guid(@"555E2D2B-EE00-47AA-AB2B-39F953F6B339");
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs:120]: This variable is assigned but not used anywhere.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs [hunk 2]
```diff
@@ -2414,45 +2414,40 @@ private static byte[] ConvertEnvVarsToByteArray(StringDictionary sd)
...
             if (_redirectstandarderror != null)
             {
+                hasRedirection = true;
                 startinfo.RedirectStandardError = true;
                 _redirectstandarderror = ResolveFilePath(_redirectstandarderror);
                 lpStartupInfo.hStdError = GetSafeFileHandleForRedirection(_redirectstandarderror, FileMode.Create);
             }
-            else
+
+            if (hasRedirection)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:2445]: Do we really need the new variable?
```suggestion
            if (startinfo.RedirectStandardInput || startinfo.RedirectStandardOutput || startinfo.RedirectStandardError)
```

File: src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs [hunk 3]
```diff
@@ -33,13 +33,12 @@ internal static void CreateRunAsAdministratorJumpList()
             {
                 try
                 {
-                    TaskbarJumpList.CreateElevatedEntry(ConsoleHostStrings.RunAsAdministrator);
+                    CreateElevatedEntry(ConsoleHostStrings.RunAsAdministrator);
                 }
-                catch (Exception exception)
+                catch (Exception)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs:38]: Could you please move unrelated changes in another PR?

**Review by iSazonov — COMMENTED** [review]
What if user redirect only output?

Can we cover the scenario we fix by tests?

File: src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs (hunk 3 above)
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs:38]: It's related. When running `Start-Process pwsh -Credential` as a non-admin user, the task bar jump list code to throw `AccessDenied` exception, and thus the `Debug.Fail` call will cause a debug build to crash, which brought trouble to the testing.

The `Debug.Fail` is not needed. The `catch (ThreadStartException)` below simply ignore the exception too.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs (hunk 2 above)
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:2445]: I thought about this and believed that using the `hasRedirection` variable can avoid unneeded member access.
Essentially, the saving in overhead doesn't really matter and I just choose one style over the other.

**Comment by daxian-dbw** [comment]
> What if user redirect only output?

When starting `pwsh` with only the output redirected, PSReadLine will not be loaded because it doesn't work when output is redirected. So, this scenario is not in our consideration for the bug.

> Can we cover the scenario we fix by tests?

Existing tests for `Start-Process` still work. The particular issue that was fixed was a pure interactive scenario, so I don't think we can cover it by test.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:2445]: I noticed this because my expectations were that this variable was doing something special, but it turned out that it wasn't. Therefore, this is puzzling.

**Review by iSazonov — APPROVED** [review]

**Comment by ltrzesniewski** [comment]
I'm pretty sure this change broke https://github.com/atuinsh/atuin/pull/2543 - looks like letting Windows figure out the default handle to use doesn't always work as well as expected. 😢

In that PR, I want to redirect stderr to a file while keeping stdout on the screen. After updating PowerShell to v7.5.0, nothing appears on screen anymore on Windows.

The following line now misbehaves:

```powershell
Start-Process -Wait -NoNewWindow -RedirectStandardError $resultFile.FullName -FilePath atuin -ArgumentList $argString
```

If I understand correctly, there's no easy workaround for this. You can't supply a custom output handle to `ProcessStartInfo`, you have to implement all the redirection yourself, which is not something I'm particularly happy to do in a PowerShell script. If you know a better solution, I'd be happy to hear.

**Comment by iSazonov** [comment]
@ltrzesniewski Please open new issue with simple repro steps.

**Comment by ltrzesniewski** [comment]
Sure, sorry. Reported here: https://github.com/PowerShell/PowerShell/issues/24986
